purchase_order_restriction/tests/test_purchase_order_restriction.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestPurchaseOrderRestriction(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestPurchaseOrderRestriction, cls).setUpClass()
+        purchase_user_group = cls.env.ref("purchase.group_purchase_user")
+        purchase_manager_group = cls.env.ref("purchase.group_purchase_manager")
+        # User in group_purchase_user
+        cls.user1 = cls._create_user(
+            cls, "Purchase User 1", "user 1", "user1@example.com", purchase_user_group
+        )
+        cls.user2 = cls._create_user(
+            cls, "Purchase User 2", "user 2", "user2@example.com", purchase_user_group
+        )
+        # Purchase order manager
+        cls.user_po_manager = cls._create_user(
+            cls,
+            "Purchase Manager",
+            "po_manager",
+            "po_manager@example.com",
+            purchase_manager_group,
+        )
+        # Partner for the POs
+        cls.partner_po = cls.env["res.partner"].create({"name": "PO Partner"})
+        # Purchase Order
+        cls._create_order(cls, "Purchaser Order 1", cls.partner_po)
+        cls._create_order(cls, "Purchaser Order 2", cls.partner_po, cls.user1.id)
+        cls._create_order(cls, "Purchaser Order 3", cls.partner_po, cls.user2.id)
+
+    def _create_user(self, name, login, email, group):
+        return (
+            self.env["res.users"]
+            .with_context(no_reset_password=True)
+            .create(
+                {
+                    "name": name,
+                    "login": login,
+                    "email": email,
```

Comment:
`email` can be fixed as it doesn't matter.

```suggestion
                    "email": "test@yourcompany.com",
```